soc/nordic/nrf54h/power.h

```diff
@@ -10,11 +10,29 @@
 #ifndef _ZEPHYR_SOC_ARM_NORDIC_NRF_POWER_H_
 #define _ZEPHYR_SOC_ARM_NORDIC_NRF_POWER_H_
 
+#include <zephyr/sys/slist.h>
+
 /**
  * @brief Perform a power off.
  *
  * This function performs a power off of the core.
  */
 void nrf_poweroff(void);
 
+/**
+ * @brief Prepare to the idle state.
+ *
+ * This function applies the required PM configuration before entering the idle state.
+ */
+void pm_state_enter_idle_prepare(void);
+
+/**
+ * @brief Get the ID of the node used by PM module.
+ *
+ * This function returns the node ID to be shared with other dependent modules.
+ *
+ * @return The pointer to the node assigned to PM module.
+ */
+sys_snode_t *pm_sys_node_id_get(void);
```

Comment:
This function does not seem to be needed anymore. It looks a bit like a hack actually.